### PR TITLE
Remove treat support

### PR DESCRIPTION
**Pull request by askoufis**
Getting the ball rolling on Braid v32. This PR is being merged into the `v32` branch, a feature branch that will eventually contain all the changes for the next major release. A feature branch makes sense here as there will be lots of code movement when migrating to crackle, so being able to review that in isolation from the treat support removal should hopefully be useful.

This PR is probably easier to review commit-by-commit.

### CODE REVIEW

File: docs/treat to vanilla-extract migration.md [hunk 1]
```diff
@@ -3,6 +3,8 @@
 This document provides guidance for consumers looking to migrate Braid-based styles from [treat](https://seek-oss.github.io/treat) to [vanilla-extract.](http://vanilla-extract.style)
 
 Since they use different file extensions (`.css.ts` vs `.treat.ts`), we're able to ease the migration by supporting both approaches simultaneously in the same project.
+However, Braid v31 was the last version to support treat styles in addition to vanilla-extract styles.
+From v32 onwards, treat styles are no longer supported and should be migrated to vanilla-extract.
```
Comment (askoufis) [inline docs/treat to vanilla-extract migration.md:7]: This reads a bit weird, even in the context of the paragraph. Will re-write this.

File: .changeset/new-cheetahs-build.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+'braid-design-system': major
+---
+
+Remove treat support
+
+[Treat](https://seek-oss.github.io/treat/) is deprecated and has been superceded by [Vanilla Extract](https://vanilla-extract.style/).
+`.treat.ts` files can no longer be used to for styling and should be converted to `.css.ts` (vanilla extract) files.
```
Comment (mrm007) [inline .changeset/new-cheetahs-build.md:8]:
```suggestion
[Treat](https://seek-oss.github.io/treat/) is deprecated and has been superseded by [vanilla-extract](https://vanilla-extract.style/).
`.treat.ts` files can no longer be used to for styling and should be converted to `.css.ts` (vanilla-extract) files.
```

_vanille-extract_ seems to be the preferred spelling (as per [`migration.md`](https://github.com/seek-oss/braid-design-system/blob/remove-treat-support/docs/treat%20to%20vanilla-extract%20migration.md))

And on _supercede/-sede_: https://writingexplained.org/supercede-vs-supersede-difference

File: packages/braid-design-system/lib/themes/makeVanillaTheme.ts [hunk 3]
```diff
@@ -110,6 +111,4 @@ export default (braidTokens: BraidTokens) => {
     transform: tokens.transforms,
     shadow: tokens.shadows,
   } as const;
-
-  return resolvedTokens;
```
Comment (mrm007) [inline packages/braid-design-system/lib/themes/makeVanillaTheme.ts:114]: Might be useful to keep the constant for debugging.

**Review by mrm007 — COMMENTED** [review]
Nice one 👍

**Review by mrm007 — APPROVED** [review]

**Review by michaeltaranto — APPROVED** [review]
👋 🍬

File: packages/generate-component-docs/src/__snapshots__/contract.test.ts.snap [hunk 4]
```diff
@@ -1930,17 +1930,7 @@ exports[`BraidProvider 1`] = `
...
-        color: { foreground: { brandAccentLight: string; brandAccent: string; cautionLight: string; caution: string; criticalLight: string; critical: string; formAccentLight: string; formAccent: string; infoLight: string; ... 14 more ...; secondaryInverted: string; }; background: { ...; }; }
-        displayName: string
-        name: string
-        space: { grid: number; space: { gutter: number; xxsmall: number; xsmall: number; small: number; medium: number; large: number; xlarge: number; xxlarge: number; }; }
-        treatTheme: string
-        vanillaTheme: string
-        webFonts: { linkTag: string; }[]
-    }
+    theme: { vanillaTheme: string; name: string; displayName: string; background: { lightMode: string; darkMode: string; }; webFonts: { linkTag: string; }[]; space: { grid: number; space: { gutter: number; xxsmall: number; ... 5 more ...; xxlarge: number; }; }; color: { ...; }; backgroundLightness: { ...; }; }
```
Comment (askoufis) [inline packages/generate-component-docs/src/__snapshots__/contract.test.ts.snap:1933]: Seems like type normalization is doing something _different_ here for some reason. Nothing really changed, other than the removal of `treatTheme`. Don't think it's anything worth worrying about.

**Pull request merged**